### PR TITLE
fix: Update state with earlier charge complete and arrival times

### PR DESCRIPTION
**Pull request by InTheDaylight14**
Added absolute value around the seconds comparison to allow for a state change when the charge complete time and the arrival time move earlier than before. 

closes https://github.com/alandtse/tesla/issues/565

### CODE REVIEW

File: tests/test_sensor.py [hunk 1]
```diff
@@ -260,6 +260,17 @@ async def test_time_charge_complete_charging(
...
 
+    car_mock_data.VEHICLE_DATA["charge_state"]["time_to_full_charge"] = 0.15
+    earlier_charge_complete = mock_now + timedelta(
+        hours=float(car_mock_data.VEHICLE_DATA["charge_state"]["time_to_full_charge"])
+    )
+    earlier_charge_complete_str = datetime.strftime(
+        earlier_charge_complete, "%Y-%m-%dT%H:%M:%S+00:00"
+    )
+
+    state = hass.states.get("sensor.my_model_s_time_charge_complete")
```
Comment (InTheDaylight14) [inline tests/test_sensor.py:271]: This is pulling the same state from line 261. Maybe HA does not update the state between the lines or I'm not changing the correct values for HA to set a new state.

**Comment by InTheDaylight14** [comment]
I've confirmed this change works in home assistant. I'm not sure how to get the state to update in the test after the times move earlier.

**Comment by InTheDaylight14** [comment]
> I've confirmed this change works in home assistant. I'm not sure how to get the state to update in the test after the times move earlier.

Such a small logic change I think we are good w/o specific test

**Comment by alandtse** [comment]
My preference is that we have a test to help with regressions. HA should have loads of examples of testing around mocking time. I think they use [timemachine](https://pypi.org/project/time-machine/) but there are alternatives.

**Comment by InTheDaylight14** [comment]
I'm able to mock the time okay. The entity state does not update after the mocked time changes (I tired to increase and decrease the time to make sure, increasing works under the original logic). Is there a way to force HA to update the sensors?

Not working test sequence:
- Check sensor state: passes
- Change Tesla API value to arrive 3 minutes earlier than before
- Set time 2 minutes in future
- Re-check state: state has not changed from the first check. Should be 1 minute before the first check.

**Comment by alandtse** [comment]
I haven't done HA testing for a while, but again, this is probably something done in core a lot.

**Comment by InTheDaylight14** [comment]
I've dumped way too much time into this, I can't solve it. 

I think I've narrowed down to, the mock controller does not grab new data and update the HA states during the test. Its not an issue with the changing time, that would be easy.

**Comment by alandtse** [comment]
Well, if you can't solve it just merge up to dev and I'll get you in the next release.

**Pull request merged**